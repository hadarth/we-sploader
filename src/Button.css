.button {
  border: 1px solid black;
  padding: 2px;
  box-shadow: 0 1px #999;
  border-radius: 4px;
}

.button:hover {
  background-color: #efefef;
  cursor: pointer;
}

.button:active {
  background-color: #569ff6;
  box-shadow: 0 1px #666;
  transform: translateY(1px);
}
