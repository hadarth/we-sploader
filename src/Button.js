import React from 'react';
import './Button.css';

const Button = ({text, onClick}) => (
  <div className="button" onClick={onClick}>
    {text}
  </div>
);

export default Button;
