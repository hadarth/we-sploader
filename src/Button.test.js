import React from 'react';
import Button from './Button';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import enzymeSerializer from 'enzyme-to-json/serializer';
expect.addSnapshotSerializer(enzymeSerializer)


Enzyme.configure({ adapter: new Adapter() });

const myclick = jest.fn()

describe("button click", () => {
  test('button should be rendered', () => {
    const res = shallow(<Button onClick={myclick} text={'text'}/>)
    expect(res).toMatchSnapshot()
  });

  test('should click', () => {
    const res = shallow(<Button onClick={myclick} text={'text'}/>)
    res.simulate('click');
    expect(myclick.mock.calls).toMatchSnapshot()
  })
});
