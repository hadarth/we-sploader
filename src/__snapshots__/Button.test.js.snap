// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button click button should be rendered 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  text
</div>
`;

exports[`button click should click 1`] = `
Array [
  Array [],
]
`;
